ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/avdtestbase.py

```diff
@@ -43,6 +43,162 @@ def render(self) -> dict:
+        except AristaAvdMissingVariableError:
+            self.log_message(key=key, logging_level=logging_level)
+            return None
+
+    def validate_data(
+        self,
+        data: dict | None = None,
+        data_path: str | None = None,
+        host: str | None = None,
+        required_keys: str | list[str] | None = None,
+        Logging_level: str | None = None,
+        **kwargs,
+    ) -> bool:
+        """
+        Validates data based on given requirements such as expected key-value pairs and required keys.
+
+        Args:
+            data (dict | None): A data dictionary to be validated. Defaults to the hostvars of the device running the test.
+            data_path (str | None): The data path in dot notation. Used for logging purposes. Index or primary key can be used for lists.
+            host (str | None): The host from which data should be retrieved. Defaults to the device running the test.
+            required_keys (str | list[str] | None): The keys that are expected to be in the data.
+            logging_level (str): Overwrites all default logging levels within this function.
+                                If not provided, the default logging level is 'WARNING' when a key is missing and 'INFO' when his value is not matching.
+            **kwargs: Expected key-value pairs in the data.
+
+        Returns:
+            bool: True if the data meets all validation requirements, otherwise False.
+
+        Example:
+            >>> validate_data(data={"a": 1, "b": 2}, data_path="some.path", required_keys=["a", "b"], c=3)
+
+            In this case, the function will log a warning message because the key 'c' with value '3' is not found,
+            and it will return False as the data doesn't meet all validation requirements.
+        """
+        host = host or self.device_name
+        data = data or get(self.hostvars, host, {})
+
+        valid = True
+
+        # Check the expected key/value pairs first
+        for key, value in kwargs.items():
+            actual_value = get(data, key)
+            if actual_value is None:
+                self.log_message(key=key, value=value, key_path=data_path, is_missing=True, logging_level=Logging_level or "WARNING")
+                valid = False
+            elif actual_value != value:
+                self.log_message(key=key, value=value, key_path=data_path, is_missing=False, logging_level=Logging_level or "INFO")
+                valid = False
+
+        # Return False if any of the expected values are missing or not matching
+        if not valid:
+            return False
+
+        # Check the required keys
+        if required_keys:
+            required_keys = [required_keys] if isinstance(required_keys, str) else required_keys
+            for key in required_keys:
+                if get(data, key) is None:
+                    self.log_message(key=key, key_path=data_path, is_missing=True, logging_level=Logging_level or "WARNING")
```

Comment:
```suggestion
                    self.log_message(key=key, key_path=data_path, is_missing=True, logging_level=logging_level or "WARNING")
```

Reply:
Fixed in 7c1821824bc682cfd9015747f8841e2e8f84991f